packages/mui-base/src/Popover/Root/usePopoverRoot.ts

```diff
@@ -0,0 +1,223 @@
+
+      if (isFocusOpen || isDismissClose) {
+        setInstantTypeState(isFocusOpen ? 'focus' : 'dismiss');
+      } else {
+        setInstantTypeState(undefined);
+      }
+
+      if (!keepMounted && !openValue) {
+        if (animated) {
+          runOnceAnimationsFinish(() => setMounted(false));
+        } else {
+          setMounted(false);
+        }
+      }
+
+      setOpen(openValue, eventValue, reasonValue);
+    },
+  });
+
+  const { delay: groupDelay, isInstantPhase, currentId } = useDelayGroup(context);
+  const openGroupDelay = typeof groupDelay === 'object' ? groupDelay.open : groupDelay;
+  const closeGroupDelay = typeof groupDelay === 'object' ? groupDelay.close : groupDelay;
+
+  let instantType = isInstantPhase ? ('delay' as const) : instantTypeState;
+  if (!open && context.floatingId === currentId) {
+    instantType = instantTypeState;
+  }
+
+  const computedRestMs = delayType === 'rest' ? openGroupDelay || delayWithDefault : undefined;
+  let computedOpenDelay: number | undefined = delayType === 'hover' ? delayWithDefault : undefined;
+  let computedCloseDelay: number | undefined = closeDelayWithDefault;
+
+  if (delayType === 'hover') {
+    if (delay == null) {
+      computedOpenDelay =
+        groupDelay === 0
+          ? // A provider is not present.
+            delayWithDefault
+          : // A provider is present.
+            openGroupDelay;
+    } else {
+      computedOpenDelay = delay;
+    }
+  }
+
+  // A provider is present and the close delay is not set.
+  if (closeDelay == null && groupDelay !== 0) {
+    computedCloseDelay = closeGroupDelay;
+  }
+
+  const hover = useHover(context, {
+    enabled: openOnHover,
+    mouseOnly: true,
+    move: false,
+    handleClose: safePolygon(),
+    restMs: computedRestMs,
+    delay: {
+      open: computedOpenDelay,
+      close: computedCloseDelay,
+    },
+  });
+  const click = useClick(context);
+  const dismiss = useDismiss(context);
+  const role = useRole(context);
+  const clientPoint = useClientPoint(context, {
+    enabled: openOnHover && followCursorAxis !== 'none',
+    axis: followCursorAxis === 'none' ? undefined : followCursorAxis,
+  });
+
+  const { getReferenceProps: getTriggerProps, getFloatingProps } = useInteractions([
+    hover,
+    click,
+    dismiss,
+    clientPoint,
+    role,
+  ]);
+
+  const getPopupProps = React.useCallback(
+    (externalProps = {}) => {
+      return mergeReactProps<'div'>(getFloatingProps(externalProps), {
+        'aria-labelledby': titleId,
+        style: {
+          // <Popover.Arrow> must be relative to the inner popup element.
+          position: 'relative',
+        },
+      });
+    },
+    [titleId, getFloatingProps],
+  );
+
+  const getTitleProps = React.useCallback(
+    (externalProps = {}) => {
+      return mergeReactProps<'h2'>(externalProps, {
+        id: titleId,
+      });
+    },
+    [titleId],
+  );
+
+  const getCloseProps = React.useCallback(
```

Comment:
We decided to adopt the hook per subcomponent pattern:
https://www.notion.so/mui-org/base-ui-Hooks-design-1b08d0b8bac94beead28f0c5736d70e3?pvs=4#a71145316649464e88d4b51acf997b12

In cases like this, it could be especially handy, as we can create a single hook and use it in Popover, Dialog, AlertDialog, and possibly others. With the ability to pass all the parameters explicitly (as per https://www.notion.so/mui-org/base-ui-Hooks-design-1b08d0b8bac94beead28f0c5736d70e3?pvs=4#4ba5108a83fd406ca834362ddb094fce), we don't have to rely on the correct context being present.